src/observation.c

```diff
@@ -220,6 +222,45 @@ u8 make_propagated_sdiffs(u8 n_local, navigation_measurement_t *m_local,
   return n;
 }
 
+/* Checks to see if any satellites have had their lock counter values have
+ * changed.
+ *
+ * If the lock counter changes it indicates that the satellite should be
+ * reinitialized in the filter. This function checks the current lock counter
+ * values in a set of single difference observations with a stored state
+ * (`lock_counters`).
+ *
+ * It outputs a the number of satellites which have had their lock counter
+ * change and a list of the corresponding PRNs.
+ *
+ * The lock counter values are then updated to reflect their changed values.
+ *
+ * \param n_sds        Number of single difference observations passed in
+ * \param sds          Array of single difference observations
+ * \param lock_counter Array of lock counter values, indexed by PRN
+ * \param sats_to_drop Output array of PRNs for which the lock counter value
+ *                     has changed
+ * \return Number of sats with changed lock counter
+ */
+u8 check_lock_counters(u8 n_sds, const sdiff_t *sds, u16 *lock_counters,
+                       u8 *sats_to_drop)
+{
+  assert(sds != NULL);
+  assert(lock_counters != NULL);
+  assert(sats_to_drop != NULL);
+
+  u8 num_sats_to_drop = 0;
+  for (u8 i = 0; i<n_sds; i++) {
+    u8 prn = sds[i].prn;
+    u16 new_count = sds[i].lock_counter;
+    if (new_count != lock_counters[prn]) {
+      sats_to_drop[num_sats_to_drop++] = prn;
```

Comment:
No objection.  I think it's pretty clear what's going on.  Expanding it will just clutter up the code.